analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/HardcodedIpAddressBase.cs

```diff
@@ -34,59 +34,61 @@ public abstract class HardcodedIpAddressBase<TSyntaxKind, TLiteralExpression> :
+            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer, CheckForHardcodedIpAddresses, Language.SyntaxKind.StringLiteralExpression);
+
+        protected bool IsHardcodedIp(string literalValue, SyntaxNode node) =>
+            literalValue != IPv4Broadcast
+            && IsNotAnObjectIdentifier(literalValue)
+            && IsRoutableNonLoopbackIPAddress(literalValue, out var address)
+            && (address.AddressFamily != AddressFamily.InterNetwork
+                || literalValue.Count(x => x == '.') == IPv4AddressParts - 1)
+            && (!(GetAssignedVariableName(node) is { } variableName)
+                || !ignoredVariableNames.Any(x => variableName.IndexOf(x, StringComparison.InvariantCultureIgnoreCase) >= 0))
+            && !HasAttributes(node);
 
         private void CheckForHardcodedIpAddresses(SyntaxNodeAnalysisContext context)
         {
-            if (!IsEnabled(context.Options))
+            if (IsEnabled(context.Options)
+                && (TLiteralExpression)context.Node is var stringLiteral
+                && GetValueText(stringLiteral) is var literalValue
+                && IsHardcodedIp(literalValue, stringLiteral))
             {
-                return;
+                context.ReportIssue(Diagnostic.Create(Rule, stringLiteral.GetLocation(), literalValue));
             }
+        }
 
-            var stringLiteral = (TLiteralExpression)context.Node;
-            var literalValue = GetValueText(stringLiteral);
+        private static bool IsRoutableNonLoopbackIPAddress(string literalValue, out IPAddress ipAddress) =>
+            IPAddress.TryParse(literalValue, out ipAddress)
+            && !IPAddress.IsLoopback(ipAddress)
+            && !ipAddress.GetAddressBytes().All(x => x == 0); // Nonroutable 0.0.0.0 or 0::0
 
-            if (literalValue == IPv4Broadcast
-                || literalValue.StartsWith("2.5.")                                  // Looks like OID
-                || !IPAddress.TryParse(literalValue, out var address)
-                || IPAddress.IsLoopback(address)
-                || address.GetAddressBytes().All(x => x == 0)                       // Nonroutable 0.0.0.0 or 0::0
-                || (address.AddressFamily == AddressFamily.InterNetwork
-                    && literalValue.Count(x => x == '.') != IPv4AddressParts - 1)
-                || (GetAssignedVariableName(stringLiteral) is { } variableName
-                    && ignoredVariableNames.Any(x => variableName.IndexOf(x, StringComparison.InvariantCultureIgnoreCase) >= 0))
-                || HasAttributes(stringLiteral))
-            {
-                return;
-            }
-
-            context.ReportIssue(Diagnostic.Create(rule, stringLiteral.GetLocation(), literalValue));
-        }
+        private static bool IsNotAnObjectIdentifier(string literalValue)
+            => !literalValue.StartsWith(OIDPrefix);   // Looks like OID
```

Comment:
I see two negations here: `IsNotAnObjectIdentifier` and `!literalValue.StartsWith(OIDPrefix)`. And the caller has the negation anyway in the name. So it's a total of three negations.

you would reduce this from three negations to a single negation in the overall logic making this method positive - `IsAnObjectIdentifier` and only leaving the negation on the caller 

```
literalValue != IPv4Broadcast
&& !IsObjectIdentifier(literalValue)

```